securedrop/journalist.py

```diff
@@ -574,23 +574,37 @@ def download_unread(sid):
     return bulk_download(sid, docs)
 
 
+@app.route('/download_all_unread')
+@login_required
+def download_all_unread():
+    docs = Submission.query.filter(Submission.downloaded == False).all()
+    if docs == []:
+        flash("No unread documents in collections!", "error")
+        return redirect(url_for('index'))
+    return bulk_dl('all_unread_' + g.user.username, docs)
+
+
 @app.route('/bulk', methods=('POST',))
 @login_required
 def bulk():
     action = request.form['action']
 
     doc_names_selected = request.form.getlist('doc_names_selected')
-    selected_docs = [doc for doc in g.source.collection
-                     if doc.filename in doc_names_selected]
-
+    if action == 'download_all':
+        selected_docs = g.source.collection
+    else:
+        selected_docs = [doc for doc in g.source.collection
+                         if doc.filename in doc_names_selected]
     if selected_docs == []:
         if action == 'download':
             flash("No collections selected to download!", "error")
+        elif action == 'download_all':
+            flash("No collections available to download!", "error")
         elif action == 'delete' or action == 'confirm_delete':
             flash("No collections selected to delete!", "error")
         return redirect(url_for('col', sid=g.sid))
 
-    if action == 'download':
+    if action == 'download' or action == 'download_all':
```

Comment:
It's a little more Pythonic to write:

```py
if action in ('download', 'download_all'):
    # ...
```

but this approach is fine since you only have 2 values to check. If you had any more than 2, I would strongly prefer the "in-tuple" pattern.